pkg/mgmt/lvmnode/lvmnode.go

```diff
@@ -54,13 +56,18 @@ func (c *NodeController) syncHandler(key string) error {
 // LVMNode
 func (c *NodeController) syncNode(namespace string, name string) error {
 	// Get the node resource with this namespace/name
-	cachedNode, err := c.NodeLister.LVMNodes(namespace).Get(name)
+	cachedNode, err := c.NodeLister.Namespace(namespace).Get(name)
 	if err != nil && !k8serror.IsNotFound(err) {
 		return err
 	}
+
 	var node *apis.LVMNode
 	if cachedNode != nil {
-		node = cachedNode.DeepCopy()
+		nodeStruct, ok := c.getStructuredObject(cachedNode)
+		if !ok {
+			return err
+		}
```

Comment:
Returning error which may not be related to this is not a good idea. We can return our own error stating that the unstructed object is not what it is intended to be. Something on these lines could be helpful rather than using some previous error stored in the err object.

Reply:
Something like this:
```runtime.HandleError(fmt.Errorf("Couldn't get node object %#v", newNode))
 		return
```

Reply:
Incorporated